server/src/main/java/org/opensearch/identity/ScheduledJobIdentityManager.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity;
+
+import org.opensearch.identity.tokens.AuthToken;
+
+/**
+ * An interface with methods used to provide security for scheduled jobs
+ *
+ * @opensearch.experimental
+ */
+public interface ScheduledJobIdentityManager {
+
+    /**
+     * Method implemented by an identity plugin to store user information for a scheduled job
+     * @param jobId The id of the scheduled job
+     * @param indexName The index where scheduled job details is stored
+     */
+    void saveUserDetails(String jobId, String indexName);
+
+    /**
+     * Method implemented by an identity plugin to delete user information for a scheduled job
+     * @param jobId The id of the scheduled job
+     * @param indexName The index where scheduled job details is stored
+     */
+    void deleteUserDetails(String jobId, String indexName);
```

Comment:
This would be if a user deletes a scheduled job altogether. If the original job details is purged from the index that stores the job details then this hook provides a way for an IdentityPlugin to hook into that and react accordingly - for the Security plugin that would mean deleting the entry from the index containing all scheduled jobs and associated user.